exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -247,6 +247,75 @@ class EntityIDColumnType<T : Comparable<T>>(
     override fun hashCode(): Int = 31 * super.hashCode() + idColumn.hashCode()
 }
 
+/**
+ * An interface defining the transformation between a database column type and a real type.
+ *
+ * @param TReal The real type to which the column value is transformed.
```

Comment:
We need to define the `Real` type. It's not clear what it is in the air.

I would also check what names are used in applications for that. There are already some known, like `Domain`, `Model`, ...

Reply:
I took these names from the existing `ColumnWithTransform` class. But we may think about complete renaming of these types. What do you think about `TOuter/TInner` (and methods `toOuter()/toInner()` respectively), or `TTransformed/TOriginal`, or `TWrapper/TColumn` or something different?